src/js/login.js

```diff
@@ -11,6 +11,40 @@ lg.loadLogin = function () {
     });
 }
 
+lg.googleLogin = function (gToken) {
+    var cellUrl = sessionStorage.googleLoginCell;
+    $.ajax({
+        type: "POST",
+        url: cellUrl + "__token",
+        data: {
+            grant_type: 'urn:x-personium:oidc:google',
+            id_token: gToken
+        },
+        headers: {
+            'Accept': 'application/json',
+            'Content-Type': 'application/x-www-form-urlencoded'
+        }
+    }).done(function (data) {
+        data.cellUrl = cellUrl;
+        var i = cellUrl.indexOf("/"); // first slash
+        i = cellUrl.indexOf("/", i + 2);  // second slash
+        data.baseUrl = cellUrl.substring(0, i + 1);
+        data.profile = JSON.parse(sessionStorage.getItem("myProfile"));
+        //data.userName = username;
+        data.userName = "googleAccount";
+        var pos = location.href.indexOf("id_token");
+        if (pos >= 0) {
+            // Exclude conjunctions that preceded id_token
+            data.logoutUrl = location.href.substring(0, pos - 1);
+        }
+        
+        sessionStorage.setItem("sessionData", JSON.stringify(data));
+        location.href = data.logoutUrl;
```

Comment:
Please use data.cellUrl

Reply:
Since data.cellUrl is the URL of cell, it does not contain information after #. If cellUrl is used, information such as mode will be lost.

Reply:
I meant it can be as simple as the following.

    var cellUrl = sessionStorage.googleLoginCell;
    ...
    data.logoutUrl = cellUrl;
    ...
    location.href = cellUrl;